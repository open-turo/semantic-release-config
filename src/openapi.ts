import {
  createPreset,
  semanticReleaseGit,
  SemanticReleasePlugin,
} from "~/_config";

const openApiSpecGlob = "spec/**/openapi.yaml";

const semanticReleaseOpenApi = (
  apiSpecFiles: string[]
): SemanticReleasePlugin => {
  return [
    "@aensley/semantic-release-openapi",
    {
      apiSpecFiles,
    },
  ];
};

/**
 * Semantic release configuration preset for OpenAPI projects. It adds the gradle-semantic-release-plugin,
 * the semantic-release-openapi plugin, and ensures that the gradle.properties file and swagger spec gets committed
 */
export = createPreset([
  // Ideally we would be able to use some sort of composite preset since OpenAPI doesn't necessarily mean gradle.
  // But we use the OpenAPI gradle plugin to generate specs and gradle to publish artifacts so this plugin needs to be defined here.
  "gradle-semantic-release-plugin",
  semanticReleaseOpenApi([openApiSpecGlob]),
  semanticReleaseGit(["gradle.properties", "README.md", openApiSpecGlob]),
]);
