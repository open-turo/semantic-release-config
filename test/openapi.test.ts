import openapi from "~/openapi";

describe("openapi", () => {
  test("adds gradle-semantic-release-plugin, semantic-release/git, and semantic-release-openapi plugins", () => {
    const gradlePlugin = openapi.plugins[3];
    const openApiPlugin = openapi.plugins[4];
    const semanticReleasePlugin = openapi.plugins[5];
    expect(gradlePlugin).toMatchSnapshot();
    expect(openApiPlugin).toMatchSnapshot();
    expect(semanticReleasePlugin).toMatchSnapshot();
  });
});
