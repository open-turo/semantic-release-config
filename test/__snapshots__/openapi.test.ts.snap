// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`openapi adds gradle-semantic-release-plugin, semantic-release/git, and semantic-release-openapi plugins 1`] = `"gradle-semantic-release-plugin"`;

exports[`openapi adds gradle-semantic-release-plugin, semantic-release/git, and semantic-release-openapi plugins 2`] = `
[
  "@aensley/semantic-release-openapi",
  {
    "apiSpecFiles": [
      "spec/**/openapi.yaml",
    ],
  },
]
`;

exports[`openapi adds gradle-semantic-release-plugin, semantic-release/git, and semantic-release-openapi plugins 3`] = `
[
  "@semantic-release/git",
  {
    "assets": [
      "gradle.properties",
      "README.md",
      "spec/**/openapi.yaml",
    ],
    "message": "ci(release): \${nextRelease.version} <% nextRelease.channel !== 'next' ? print('[skip ci]') : print('') %>

\${nextRelease.notes}",
  },
]
`;
